interval.hh

```diff
@@ -462,27 +459,35 @@ public:
         return std::move(_interval);
     }
 
+    template<typename Comparator>
+    bool empty(Comparator&& cmp) const {
+        return _interval.is_wrap_around(cmp);
+    }
+
```

Comment:
Should we allow constructing ranges with lower > upper?

Options:
 1. yes
 2. disallow in constructor (throw or assert, or trust user), add separate free function that checks for the condition and constructs a real empty instead

Reply:
In order to disallow in constructor we need an access to a comparator in constructor - not always the case, especially with IDL.

Reply:
I meant "logically disallow".